unique_sdk/unique_sdk/_api_requestor.py

```
@@ -354,14 +354,15 @@ def interpret_response(
 
     def handle_error_response(self, rbody, rcode, resp, rheaders) -> NoReturn:
         try:
-            error_data = resp["error"]
-        except (KeyError, TypeError):
+            error_data = resp.get("error")
+        except TypeError as e:
```

This would now return None if the key "error" does not exist. It will also never throw a TypeError. The old code seems to be correct. I wonder if  there is anything to do on our sides besides explaining the behavior?

---

if the `error` is `None` we would be able to to handle it in the next lines of code as seen here...https://github.com/Unique-AG/ai/blob/37bd5caa6da19a057436db5487c4ac945be7b758/unique_sdk/unique_sdk/_api_requestor.py#L373-L375

---

If we take care of the key_error to fail silently, it would still show the message requested, currently from the error_log this is the body of their message 

```
'{\n  "message":"The upstream server is timing out",\n  "request_id":"528dfd3c8968de628314af76284b4e11"\n}'
```

it doesn't have the required `error` parameter.

---

Yes. I wasn't really sure what the prior writer was trying to catch with the `TypeError` perhaps there are times that the `resp` wouldn't be a dictionary?